crates/types/src/config/common.rs

```diff
@@ -497,44 +496,65 @@ pub enum LogFormat {
     Json,
 }
 
-/// # Service Client options
+/// # Metadata client options
 #[serde_as]
 #[derive(Debug, Clone, Serialize, Deserialize, derive_builder::Builder)]
 #[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
 #[cfg_attr(
     feature = "schemars",
-    schemars(rename = "MetadataStoreClientOptions", default)
+    schemars(rename = "MetadataClientOptions", default)
 )]
 #[builder(default)]
-#[serde(rename_all = "kebab-case")]
-pub struct MetadataStoreClientOptions {
-    /// # Metadata store
+#[serde(rename_all = "kebab-case", default)]
+pub struct MetadataClientOptions {
+    /// # Metadata client type
     ///
-    /// Metadata store server kind.
-    pub metadata_store_client: MetadataStoreClient,
+    /// Which metadata client type to use for the cluster.
+    #[serde(flatten)]
+    pub kind: MetadataClientKind,
 
     /// # Connect timeout
     ///
     /// TCP connection timeout for connecting to the metadata store.
     #[serde_as(as = "serde_with::DisplayFromStr")]
     #[cfg_attr(feature = "schemars", schemars(with = "String"))]
-    pub metadata_store_connect_timeout: humantime::Duration,
+    pub connect_timeout: humantime::Duration,
 
     /// # Metadata Store Keep Alive Interval
     #[serde_as(as = "serde_with::DisplayFromStr")]
     #[cfg_attr(feature = "schemars", schemars(with = "String"))]
-    pub metadata_store_keep_alive_interval: humantime::Duration,
+    pub keep_alive_interval: humantime::Duration,
 
     /// # Metadata Store Keep Alive Timeout
     #[serde_as(as = "serde_with::DisplayFromStr")]
     #[cfg_attr(feature = "schemars", schemars(with = "String"))]
-    pub metadata_store_keep_alive_timeout: humantime::Duration,
+    pub keep_alive_timeout: humantime::Duration,
 
-    /// # Backoff policy used by the metadata store client
+    /// # Backoff policy used by the metadata client
     ///
-    /// Backoff policy used by the metadata store client when it encounters concurrent
-    /// modifications.
-    pub metadata_store_client_backoff_policy: RetryPolicy,
+    /// Backoff policy used by the metadata client when it encounters concurrent modifications.
+    pub backoff_policy: RetryPolicy,
+}
+
+impl Default for MetadataClientOptions {
+    fn default() -> Self {
+        Self {
+            kind: MetadataClientKind::Native {
+                addresses: vec![DEFAULT_ADVERTISED_ADDRESS
+                    .parse()
+                    .expect("valid metadata store address")],
+            },
+            connect_timeout: Duration::from_secs(3).into(),
```

Comment:
Perhaps you have better observations about how this behaves in practice but is 3s perhaps much too high for a connect timeout?

Reply:
Something to observe and to adjust if this does not work out. We recently reduced it from 5s to 3s.